(feat)ml: add model pipeline

alfcan opened the pull request.

alfcan committed:
feat(ml): add pipeline for NLU model

* use RoBERTa model pertained to the classification of requests
* add input testing

alfcan committed:
fix(ml): change the calculation of metrics 

* use the sklearn.metrics library
* Delete the Specifity metric and add the MCC (Matthew's Correlation Coefficient) metric

Closes #1

davide-lagamba reviewed (changes requested):
In "CADOCS_NLU.ipynb", the calculation of Accuracy, Precision and Recall could be reconsidered since the values obtained are equal between them.
Some comments should be in English instead of Italian.
Some comments partially describe the code and could be changed.

alfcan committed:
fix(ml): fix compute metrics

- change the library for compute metrics to HuggingFace Evaluate
- use augmented dataset instead of original dataset

alfcan commented:
The value of the metrics is the same since the increased dataset was used and improved performance, but the metrics are calculated differently from the previous version.

alfcan commented:
In this branch is also solved issue #2
